operator/api/redpanda/v1alpha2/node_pool_types.go

```diff
@@ -106,3 +122,17 @@ type EmbeddedNodePoolSpec struct {
 	SidecarImage               *RedpandaImage      `json:"sidecarImage,omitempty"`
 	InitContainerImage         *InitContainerImage `json:"initContainerImage,omitempty"`
 }
+
+func MinimalNodePoolSpec(cluster *Redpanda) NodePoolSpec {
+	return NodePoolSpec{
+		ClusterRef: ClusterRef{
+			Name: cluster.Name,
+		},
+		EmbeddedNodePoolSpec: EmbeddedNodePoolSpec{
```

Comment:
What are the defaulting semantics here? Do we adopt any values on the default nodepool / sts? If so, can we document that on the type itself so it appears in docs for end users?